homeassistant/components/zone/__init__.py

```diff
@@ -110,6 +110,10 @@ def async_active_zone(
     min_dist = None
     closest = None
 
+    # Update state of current_zone
+    if current_zone:
+        current_zone = hass.states.get(current_zone.entity_id)
```

Comment:
Maybe change the variable name here so we don't overwrite the variable with another type.
```suggestion
        current_zone_state = hass.states.get(current_zone.entity_id)
```

Reply:
It's actually the same type. We could instead update such that we pass in the entity_id?

Reply:
Right, I see that now. I was confused. If we always need to get the current state before we can use it, yes maybe it's enough to store the entity_id.

Reply:
If we don't refresh the state we may not leave a zone which has been been modified (for example it was set to passive or became unavailable), I guess we need to check that?

Reply:
Sounds good.